src/Microsoft.ML.HalLearners/OlsLinearRegression.cs

```diff
@@ -34,25 +34,32 @@
 namespace Microsoft.ML.Trainers.HalLearners
 {
     /// <include file='doc.xml' path='doc/members/member[@name="OLS"]/*' />
+    [BestFriend]
     public sealed class OlsLinearRegressionTrainer : TrainerEstimatorBase<RegressionPredictionTransformer<OlsLinearRegressionModelParameters>, OlsLinearRegressionModelParameters>
     {
-        public sealed class Arguments : LearnerInputBaseWithWeight
+        public sealed class Options : LearnerInputBaseWithWeight
```

Comment:
Is LearnerInputBaseWithWeights still a good name? Would it be better to use `LearnerOptionBaseWithWeight`?

Reply:
think that goes beyond the scope of this PR

---
In reply to: [248857544](https://github.com/dotnet/machinelearning/pull/2170#discussion_r248857544) [](ancestors = 248857544)